### DISCUSSION TITLE
Jellyfin for Kodi - Starting from Jellyfin 10.9 it is no longer possible to set the shared network folder.

### DISCUSSION THREAD

**Discussion by Philmo67** [jellyfin/jellyfin#11654] [Q&A] [answered]
Hello,
Jellyfin for Kodi documentation states "Starting from Jellyfin 10.9 it is no longer possible to set the shared network folder.".
Could someone explain how native (non transcoding) play will work/be possible starting from this version ?

**Comment by Shadowghost** [comment]
The files are streamed directly from the server. If you want direct file based access, you should use something like samba and not Jellyfin.

**Reply by spcano01** [reply]
But why smb?  NFS worked so much better, why the removal?!

**Reply by supertin** [reply]
I remember trying the native method when first playing with Jellyfin, and the CPU usage on the server was significantly higher. Unless that's dramatically changed, I'd drop Jellyfin altogether if they removed the shared folder thing altogether (still works for now, but no UI).

For me, Jellyfin isn't about the Netflix style streaming. It's purely to handle metadata and syncing from one place instead of having to fix up half a dozen individual Kodi installs whenever something gets named wrong or files moved. With Jellyfin, I just log into the web interface, fix the issue and it syncs to all media players.

**Reply by sic6SaNdMaN** [reply]
Exactly what supertin says.
I don't think , that my raspberry pi 4B can handle 150-200 MBit/s files over the built-in http server. The CPU load becomes too high.
NFS is no problem at all, SMB has noticable CPU load but stays fine (even up to 600 MBit/s).
But I tried AddOn-Mode once and it definitvely could'nt handle even lower bitrate files properly.

So, yes: If native paths go away, jellyfin has to go away - unfortunately.

**Reply by sic6SaNdMaN** [reply]
> Exactly what supertin says. I don't think , that my raspberry pi 4B can handle 150-200 MBit/s files over the built-in http server. The CPU load becomes too high. NFS is no problem at all, SMB has noticable CPU load but stays fine (even up to 600 MBit/s). But I tried AddOn-Mode once and it definitvely could'nt handle even lower bitrate files properly.
> 
> So, yes: If native paths go away, jellyfin has to go away - unfortunately.

Actually, forget what I wrote.
I changed all my Clients to AddOn-Mode and tested A LOT the last days.
And even with files that have 150 MBit/s bitrate the jellyfin process doesn't eat up more than 35-45% CPU (on only ONE CPU, to be clear). Some small spikes to about 60%, but they were very rare and even that isn't a problem, of course.

It seems, that the built-in webserver got WAY more efficient since my last tests when I started with jellyfin years ago.
I also overclocked the Pi and made several tweaks to improve network throughput, but that can't possible have such an enormous impact on the performance.
Overall with the current software my setup is well capable of everything I throw at it - so it won't be a problem for me, as far as i can judge this for now.

**Reply by APraxx** [reply]
Still before i had almost zero load on my jellyfin machine while playing and now i can tell.

I think it's sad to see this go as on small setups where transcoding may not be an option and where even the tiny load of the "addon mode" makes a differnce this was a easy and well working solution

**Comment by Philmo67** [comment]
Is native playback dropped with the Jellyfin for KODI plugin ? What would be the best practice in order to let KODI handle the whole potential transcoding tasks (no GPU on the Jellyfin side) ?

**Reply by sjorge** [reply]
Kodi would still retrieve the file from jellyfin, without the server transcoding it. It just won't do it using the shared network path but over HTTP(s)

**Reply by oddstr13** [reply]
jf4kodi direct plays most if not all content in addon mode too, unless you've configured limits on bitrate or codecs in the addon settings.
The main differences are that the content is fetched via the Jellyfin server, not from the shared network folder path, and the fact that I actually use the addon mode myself (aka this is the only method I test with).

**Comment by sunbiker** [comment]
oh no, since I have several NAS, this function was very important for me.

**Reply by thedreaddpirate** [reply]
You can specify more than one path per library.  You just have to mount them to a path or drive letter in the OS.

**Comment by Philmo67** [comment]
So far, the only regression I've noticed, going from native to addon mode, is when attempting to read *.disc files (to allow users to search through the list of unripped discs). Instead of a message indicating where to find the disc to play the movie (as here https://kodi.wiki/view/Media_stubs), I get a playback error.

**Reply by Shadowghost** [reply]
That is to be expected. You won't be able to "direct play" files Jellyfin can't expose: DVD/BD folder layouts, disc types

**Reply by Philmo67** [reply]
When trying to playback *.disc files through Jellyfin (web) you get a popup telling you are trying to playback a placeholder for physical media. This kind of message would avoid confusion (playback error vs physical placeholder)

**Comment by jenshenk** [comment]
So this is really a feature not a bug?! Anyway I noticed old configured folders keep working and if you e.g. edit the jellyfin/config/root/default/Music/options.xml file you can still add the NetworkPath line and keep the old functionality intact. Just added a new source and kodi likes it.

**Comment by spcano01** [comment]
Yes, just need to update the .xml directly.  Not sure if that gets overwritten on an update or not, delaying updating until I find out/make recent backup of just that folder/files.  

Not sure why this had to change, not seeing upside but just removed the easy ability for us to hardcode a different filesystem (vs. perhaps a client/OS that only allows one) and to have native access.  No reason to not have this in home where I 100% direct play everything (via Kodi), but then use Jellyfin clients outside of the home.

**Reply by oddstr13** [reply]
Only removed in the UI as of yet, but it is deprecated and will most likely be removed completely in the future.

The normal addon mode in kodi should direct play pretty much everything.

**Reply by jenshenk** [reply]
Is there a reason why this will be removed? It made it so easy when the docker container used a different way of accessing files than kodi clients and I don't really understand why such an handy feature suddenly needs to go.

**Reply by dominic330** [reply]
I use jellyfin precisely because of this function.
Kodi at home and Jellyfin app on the go. Since I have a Dune media player, I am dependent on the path specification, otherwise the external player does not know where the file is located. I hope the function will be implemented again...

**Reply by jenshenk** [reply]
I just reverted back to the 10.8 branch due to random other issues - it's a bit sad, as jellyfin in general and especially 10.8 was rock solid and compared to other stuff I use never something I needed to worry about and it's the first time I got frustrated with it. I think there might have been just too many ambitious changes towards 10.9. Note for anyone reading this: If you also do this make sure to clear your browser cache and the app's cache otherwise you'll encounter strange login issues.

**Reply by oddstr13** [reply]
Try using addon mode (the default and preferred) in Jellyfin for Kodi, the extra path setting isn't used there.

**Reply by spcano01** [reply]
We have, as it's the default one...but with high bit-rate and large filesz the performance via SMB (defacto protocol via add-on) vs NFS (option via native) was night and day.

**Comment by spcano01** [comment]
Yes, I did too but the performance difference of that vs NFS (via native)
is night and day in my scenario.

**Reply by jenshenk** [reply]
I had the same experience with my setup, native just works better. What still confuses me the most about this is that I did not yet hear about something great that is only possible if this goes away, to me it appears it's just about tiding up or something. And for that it's a pretty radical change, even if you ignore potential performance aspects, all users who need to change to add-on mode will need to reset the databases of all their devices.

**Comment by draxcp6** [comment]
I just wanted to post workaround.

Open file  `appdata/jellyfin/root/default/<LIBRARY_NAME>/options.xml` in your favorite text editor and add following line:

`<NetworkPath>\\127.0.0.1\Some\Path</NetworkPath>`
within `<MediaPathInfo>` tag. Just don't forget to replace path with your own.

So end result should be something like:

  <PathInfos>

    <MediaPathInfo>

      <Path>/Movies</Path>

      <NetworkPath>\\127.0.0.1\Some\Path\to\Movies</NetworkPath>

    </MediaPathInfo>

  </PathInfos>

But I don't know how long it will last. I will probably stay on 10.9 until this feature is brought back because performance was terrible doing it recommended way + it had some quirks with subtitles.

**Comment by spcano01** [comment]
Agreed, will check changelogs and when this changes - I can't update.
Really don't see why we're removing features, does this cause
moderate/significant dev time to maintain?

**Reply by jenshenk** [reply]
I don't have an answer to this but I get the feeling something really changed with jellyfin. Because I didn't like the hacky way of configuring the network folder but also due to 10.9 having a significantly higher memory footprint (in my case around 500mb vs 1.x gig) I decided to stay with 10.8 for now. One of my android boxes just updated the jellyfin android tv app yesterday and now I just get an error that the server needs to be updated to 10.9.9. It's all getting a bit frustrating for an otherwise really great project.

**Comment by Dnkhatri** [comment]
This will probably cause peoples home network bandwidth to get over saturated and hurt performance badly for other things apart from jellyfin also . As now jellyfin takes the files and sends to the client so using twice the bandwidth where previously the client was playing it directly.

**Comment by sic6SaNdMaN** [comment]
Should we feature-request the NativePath-Option back in?

**Comment by kennedybaird** [comment]
Considering the potential bandwidth implications (as @Dnkhatri points out) - it seems strange that this has been removed.

**Comment by spcano01** [comment]
Just to clarify, this option still works as of 4.9.11.  Removing from the UI is annoying, but as long as I can do this via options.xml and it doesn't reset on upgrading - I'm good.

**Reply by kennedybaird** [reply]
I'm having trouble finding options.xml, could you point me in the right direction?

**Reply by hexoverflow** [reply]
It is going away in 10.10 and they are already preparing for the removal from the addon too. I don't get why we are removing perfectly working features...

https://github.com/jellyfin/jellyfin-kodi/pull/925

**Reply by gnattu** [reply]
> I don't get why we are removing perfectly working features...

Because it is not working perfectly and it is causing maintenance headaches.

**Reply by draxcp6** [reply]
> I'm having trouble finding options.xml, could you point me in the right direction?

Take a look at [post](https://github.com/jellyfin/jellyfin/discussions/11654#discussioncomment-10331035).

**Reply by draxcp6** [reply]
> > I don't get why we are removing perfectly working features...
> 
> Because it is not working perfectly

Don't know what are you talking about. I am using it for past few years, and it works perfectly.

> and it is causing maintenance headaches.

But without this feature it is going to cause user headaches...

**Reply by cvium** [reply]
> > > I don't get why we are removing perfectly working features...
> 
> > 
> 
> > Because it is not working perfectly
> 
> 
> 
> Don't know what are you talking about. I am using it for past few years, and it works perfectly.
> 
> 
> 
> > and it is causing maintenance headaches.
> 
> 
> 
> But without this feature it is going to cause user headaches...

This is the only client that supports this feature, and while it has worked for you, it has caused problems for many others.

What kind of home network can't handle this traffic?

**Reply by hexoverflow** [reply]
There's multiple people saying that files do not play correctly in addon mode (including me). There's certainly an issue with addon mode and kodi which is why native mode needs to stay (as a client only option)

**Reply by bzzbrr** [reply]
> There's multiple people saying that files do not play correctly in addon mode (including me). 

Could you give a very specific example (full media info) of a file that doesn't work in addon mode that does in native?  I've been trying to document these and haven't had much luck with generic examples such as "high bitrate".

**Reply by sic6SaNdMaN** [reply]
Well it's definitely because of high bitrates.
I'm using a raspberry Pi 4B and when the bitrates are in the range of 150 MBit/s and above, the CPU load becomes too high when the built-in http server is used.
NFS and even SMB is fine (until ca. 600 MBit/s).
But I tried AddOn-Mode once and it definitvely could'nt handle even lower bitrate files (in the range of 65-80 MBit/s) properly.
This was the only reason, I switched to native paths in the first place. If it wouldn't have been neccessary, I wouldn't have bothered with this extra step.

**Reply by sic6SaNdMaN** [reply]
> Well it's definitely because of high bitrates. I'm using a raspberry Pi 4B and when the bitrates are in the range of 150 MBit/s and above, the CPU load becomes too high when the built-in http server is used. NFS and even SMB is fine (until ca. 600 MBit/s). But I tried AddOn-Mode once and it definitvely could'nt handle even lower bitrate files (in the range of 65-80 MBit/s) properly. This was the only reason, I switched to native paths in the first place. If it wouldn't have been neccessary, I wouldn't have bothered with this extra step.

Well, forget what I wrote there.
It seems that the built-in webserver got WAY more efficient.
See my other comment: https://github.com/jellyfin/jellyfin/discussions/11654#discussioncomment-11053039

**Reply by spcano01** [reply]
What's the full way to "reset" the plug-in from native to add-on?  Doesn't seem to have a way than un/reinstalling it or clearing data and readding everything.  

I can reset local database and/or repair libraries, but I don't think that allows us to change from native to add-on.

**Reply by Jucgshu** [reply]
Database reset allows to choose between native and addon mode.

**Comment by kurosh** [comment]
Anyone else have issues with JellyFin4Kodi forgetting the password, even after resettting the local database and setting to 'Addon' mode and doing a full resync? All my clients seem to forget the password every 24 hours or so I think. Server is 10.10.0.

Cheers.

**Comment by Ulrar** [comment]
Would this being removed lead to Kodi saying the file isn't available anymore, and asking to remove it from the library ?

It's not all files but most of them, which is fairly weird, since the upgrade to 10.10. Only happens in native mode so I assume it's related

**Reply by ramorous** [reply]
This started happening for me as well, but only newly added files. I gave up and went transcoding for now. I may rollback to 10.9 and revert and wait there until it's fixed or maybe eventually find something else that fits my needs.

**Comment by PrplHaz4** [comment]
I don't think it's linked above, but for anyone following here, there is some work going on to support network folders in the kodi plugin:
- https://github.com/jellyfin/jellyfin-kodi/pull/926

**Reply by sic6SaNdMaN** [reply]
> I don't think it's linked above, but for anyone following here, there is some work going on to support network folders in the kodi plugin:
> 
> * [Add client native mode path replacement jellyfin-kodi#926](https://github.com/jellyfin/jellyfin-kodi/pull/926)

Quoting myself from there:
"Well I don't see this being an upgrade in any form, as I have to configure all of these path-replacements on all of my boxes and various installations.
Instead of having a configuration centralized on the server and just working for every client, the same configuration-process has to be done repeatedly on all clients.
That's a huge regression. Especially when you concern, how tricky those paths can be and that most of the clients are TV-boxes or Streamer-boxes and you have to enter all of these paths with a remote instead of a keyboard...
Sooo... no, not an upgrade, not an upgrade by any means at all.

BUT: Nonetheless I'm happy it exists. It surely can help, when direct paths are really needed."

There are other problems/bugs with AddOn-mode, tho.
So this is and stays a huge DOWNgrade.

**Comment by merc4derp** [comment]
So, I'm late to the party and I just discovered this new brilliant feature.
What the fuck were you thinking? Enshittificating fucking Jellyfin? Are you out of your fucking minds?

BRING BACK NATIVE MODE YOU CLOWNS.

Peace out. Reverting to 10.9 and staying there until common sense is reinstated.

**Comment by sic6SaNdMaN** [comment]
Having used AddOn-Mode in Jellyfin4Kodi for the last weeks as of this discussion.
And thus I had AT LEAST the following bugs:
- https://github.com/jellyfin/jellyfin-kodi/issues/912
- https://github.com/jellyfin/jellyfin-kodi/issues/700 (extremely annoying! Switched back to native mode because of that!)

And perhaps also:
- https://github.com/jellyfin/jellyfin-kodi/issues/538
- (https://github.com/jellyfin/jellyfin-kodi/issues/932)

So... yeah... even more reasons for sticking to native mode.

**Comment by cfelicio** [comment]
In case anyone stumbles on this in 2026, you can use jellyfin mpv shim to do path substitutions now.

**Comment by oddstr13** [comment, marked as answer]
Closing as resolved, the following PR has been merged:
https://github.com/jellyfin/jellyfin-kodi/pull/938

Further problems related to this should be discussed in the respective clients repository.
Search first, then create new issues if your issue isn't already covered. Make sure to also check closed issues and open PRs.
